### ISSUE TITLE
Support for projection v2.0.0 extension

### ISSUE THREAD

**Issue by jlaura** [stac-utils/pystac#1286]
When [the switch to `proj:code`](https://github.com/stac-extensions/projection/pull/12) lands in the projection extension and a 2.0 is released, it would be awesome to have PySTAC update for the new field. 

Believe that will mean supporting projection v1.1.0 that uses `proj:epsg` and then 2.0.0 that has the new `proj:code` extension.

Best, 
Jay

**Comment by harryle95** [comment]
Hey guys, I am just wondering what's the progress on this. We are using this pystac extension for our project and found the same issue with code and epsg. I am happy to help with the PR if needed.

**Comment by KeynesYouDigIt** [comment]
@gadomski / @jsignell Should I take a crack at this? looking for a nice "medium sized" issue to keep getting my feet wet :)

**Comment by gadomski** [comment]
Yeah! @jsignell has a WIP branch here: https://github.com/stac-utils/pystac/pull/1287

**Comment by KeynesYouDigIt** [comment]
@gadomski can you assign me for now? I'll start breaking it down tomorrow, will raise heck if I get stuck :smile:

**Comment by gadomski** [comment]
You bet! Thanks for jumping in! FYI I'm off until 02 Jan so won't be doing any reviews until then.
